pluginlib/package.xml

```diff
@@ -21,12 +21,17 @@
   <buildtool_depend version_gte="0.5.68">catkin</buildtool_depend>
 
   <build_depend>cmake_modules</build_depend>
+  <build_depend>libboost-filesystem-dev</build_depend>
+  <build_depend>class_loader</build_depend>
+  <build_depend>rosconsole</build_depend>
+  <build_depend version_gte="1.11.1">roslib</build_depend>
+  <build_depend>tinyxml2</build_depend>
 
-  <depend>boost</depend>
-  <depend>class_loader</depend>
-  <depend>rosconsole</depend>
-  <depend version_gte="1.11.1">roslib</depend>
-  <depend>tinyxml2</depend>
+  <build_export_depend>libboost-filesystem-dev</build_export_depend>
```

Comment:
Does this still bring in `boost/algorithm`, `boost/bind`, `boost/foreach`, and `boost/shared_ptr`?

https://github.com/ros/pluginlib/blob/dfd49dad457d81f4fc5984a198f4b2abc39ba7c6/pluginlib/include/pluginlib/class_loader.hpp#L37

https://github.com/ros/pluginlib/blob/dfd49dad457d81f4fc5984a198f4b2abc39ba7c6/pluginlib/include/pluginlib/class_loader_imp.hpp#L50-L53

https://github.com/ros/pluginlib/blob/dfd49dad457d81f4fc5984a198f4b2abc39ba7c6/pluginlib/test/utest.cpp#L103

Reply:
Good question,

Any `libboost-LIBNAME-dev` package depends on `libboost-dev` that brings in ALL boost headers.

Then if we want to be sure these headers are still brought in the day the filesystem dependency is removed from pluginlib, and no other dependency brings boost headers in (very unlikely), we need to add a dependency on `libboost-dev` in this package.